src/mpi/romio/adio/common/onesided_aggregation.c

```diff
@@ -275,14 +275,14 @@ printf("ADIOI_OneSidedWriteAggregation started on rank %d\n",myrank);
     if (!bufTypeIsContig) {
    /* Flatten the non-contiguous source datatype and set the extent. */
       if ((stripe_parms.stripeSize == 0) || stripe_parms.firstStripedWriteCall) {
-        flatBuf = ADIOI_Flatten_and_find(datatype);
-        MPI_Type_extent(datatype, &bufTypeExtent);
+        flatBuf_global = ADIOI_Flatten_and_find(datatype);
+        MPI_Type_extent(datatype, &bufTypeExtent_global);
       }
 
 #ifdef onesidedtrace
-      printf("flatBuf->count is %d bufTypeExtent is %d\n", flatBuf->count,bufTypeExtent);
-      for (i=0;i<flatBuf->count;i++)
-        printf("flatBuf->blocklens[%d] is %d flatBuf->indices[%d] is %ld\n",i,flatBuf->blocklens[i],i,flatBuf->indices[i]);
+      printf("flatBuf->count is %d bufTypeExtent is %d\n", flatBuf_global->count,bufTypeExtent_global);
+      for (i=0;i<flatBuf_global->count;i++)
+        printf("flatBuf->blocklens[%d] is %d flatBuf->indices[%d] is %ld\n",i,flatBuf_global->blocklens[i],i,flatBuf_->indices[i]);
```

Comment:
The last instance looks wrong here in this printf. Also the first reference is not changed to `_global` when it probably should be. This is all inside an `#ifdef` so was not compile tested.

Reply:
That is correct. Thanks for catching that. These have been fixed.